### PR TITLE
Kp/greedy

### PR DESCRIPTION
**Pull request by kyra-at-ada**
**Summary of Changes**
- Greedy overview
- Dijkstra's as a greedy algorithm
    - Greedy Choice Principle
    - Optimal Substructure Principle
- Applying Greedy Algorithms (Coin Change)
- Pros & Cons of Greedy Algorithms
- Updates file names to order lessons
- Updates link to algorithm lesson in About TAA

**For Review**
- Structure & Clarity of Explanation
- Note that the video explanation of the coin change problem is still missing
- Note that the gif of dijkstras algorithm is too big to preview and reads as an error, but says it should be okay when published
- Also note that sliding window will actually go after dynamic programming (since it is a DP technique)

### CODE REVIEW

File: 07-algorithms/03-greedy-algorithms.md [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+Additionally, it is often difficult to prove that a greedy algorithm is correct. We mentioned previously that Dijkstra's is proven to provide the shortest path between two nodes in a graph. It may make sense to us that the optimal solution to the shortest subpath leads us to the optimal overall path, but how do we prove that is true? We can ask ourselves a similar question to the coin change problem. How do we prove that choosing the largest coin available leads us to the best result?
+
+In an interview setting, it is often easier to come up with counterexamples to demonstrate that a greedy algorithm won't produce an optimal result and choose to move forward with another approach. If it does seem like the greedy approach is viable, we can support our solution with thoughtful, robust tests. 
+
+<!-- available callout types: info, success, warning, danger, secondary, star  -->
+### !callout-secondary
+
+## Mathematical Proof
+
+THe optimal greedy algorithms is a proof by induction. This is beyond the scope of this course, but if you are curious, the proof for Dijkstra's can be [viewed here](https://web.engr.oregonstate.edu/~glencora/wiki/uploads/dijkstra-proof.pdf).
```
[the remark this answers is not in the document]
Comment (char-adadev) [reply 07-algorithms/03-greedy-algorithms.md:414]: totally un-related but when I read the words "proof by induction" I got some pretty terrible flashbacks to my first Logics course lmao

**Review by char-adadev — APPROVED** [review]
Looks great overall! I left a few suggestions and pointed out one inconsistency between an image and the text. Thank you for working on this! Approving as I believe we should be good to go with the suggestions applied 👍🏾

File: 07-algorithms/03-greedy-algorithms.md [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+Greedy algorithms are most often applied to optimization problems. Combined with the greedy choice and optimal substructure properties, let's take a look at how we can apply the greedy technique to a problem. 
+
+#### The Coin Change Problem 
+
+Say we have some set of different coin types and infinite supply of each coin type. We are given a target sum and would like to choose coins to total that sum. Return a list of the minimum number of coins needed to make up the target sum. If it is impossible to produce that sum with the provided coins, return an empty list. 
+
+##### Example Input 1
+![Coin Change Nominal Example](./images/coin-change-example-1.png)
+<br>
+Input: `coins = [1, 2, 4, 25, 50, 100]`, `target_sum = 156`
```
Comment (kyra-at-ada) [inline 07-algorithms/03-greedy-algorithms.md:114]:
```suggestion
Input: `coins = [1, 2, 5, 25, 50, 100]`, `target_sum = 157`
```

File: 07-algorithms/03-greedy-algorithms.md [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+Output: `[100, 50, 4, 2]`
```
Comment (kyra-at-ada) [inline 07-algorithms/03-greedy-algorithms.md:115]:
```suggestion
Output: `[100, 50, 5, 2]`
```

**Pull request merged**